vulnerabilities/data_source.py

```diff
@@ -70,9 +85,11 @@ class Advisory:
     """
     summary: str
     impacted_package_urls: Iterable[PackageURL]
-    resolved_package_urls: Iterable[PackageURL] = dataclasses.field(default_factory=list)
+    resolved_package_urls: Iterable[PackageURL] 
+    vuln_pkg_version_ranges : Iterable[VulnPkgVersionRange]
```

Comment:
I reckon you would want that `vuln_pkg_version_ranges` be optional? and replace the the `impacted_package_urls` ?
Or would we keep both?
Could you have both at the same time?
And if we have bot would these be rather names something like `impacted_package_urls_specification` , or ``impacted_package_urls_requirements` or `impacted_package_urls_version ranges` or ...?`